3004 table custom fields from csv

entantoencuanto opened the pull request:
Closes #3004


## :v: What does this PR do?

Allows admins to upload custom field from a CSV according to description of the related issue

## :mag: How should this be manually tested?

The CSV of the example in the issue has been uploaded after creating a custom field of type table with `campo-indicadores` uid and configuration:

```json
{
  "columns": [
    {
      "id": "name",
      "type": "text",
      "name_translations": {
        "ca": "Nom",
        "en": "Name",
        "es": "Nombre"
      }
    },
    {
      "id": "value",
      "type": "text",
      "name_translations": {
        "ca": "Valor",
        "en": "Value",
        "es": "Valor"
      }
    },
    {
      "id": "description",
      "type": "text",
      "name_translations": {
        "ca": "Descripció",
        "en": "Description",
        "es": "Descripción"
      }
    },
    {
      "id": "other",
      "type": "text",
      "name_translations": {
        "ca": "Otro texto",
        "en": "Other text",
        "es": "Otro texto"
      }
    }
  ]
}
```

The value field can be configured as type float, but some values as `0,5` would be converted to 0 and `10%` to 10

## :eyes: Screenshots

### Before this PR

🚫 

### After this PR

![3004_after](https://user-images.githubusercontent.com/446459/81851214-65287680-9559-11ea-8e2b-6ec7ae1c7d8f.gif)


## :shipit: Does this PR changes any configuration file?

No

(Changes in these files might need to update the role in Ansible)

## :book: Does this PR require updating the documentation?

Pending

entantoencuanto committed:
Add action and form object to import plan table custom fields data from a CSV

entantoencuanto committed:
change fixture custom_field uid

entantoencuanto committed:
Fix and add integration tests

entantoencuanto committed:
Make case insensitive columns lookup in table custom fields csv import

entantoencuanto committed:
Allow indicators custom field tables not using vocabularies

ferblape commented:
> The value field can be configured as type float, but some values as 0,5 would be converted to 0 and 10% to 10

Why is this an issue? Why not `.tr(',', '').to_f`?

ferblape reviewed (changes requested):
I think you should deal properly with floats

entantoencuanto committed:
Parse strings to floats in custom fields including commas

entantoencuanto committed:
Add tests to numeric custom field value parsing

entantoencuanto commented:
Changes added to parse better the strings:

* If a string contains more than a "," or at least a ".", the commas are removed and the value is transformed using `to_f`
* If the string contains only a "," is replaced by a "." and transformed using `to_f`

These changes affect also to the way a custom field of type numeric parses the values received

ferblape commented:
In https://diba-plan.gobify.net/admin/plans/41/import_csv I'm importing this CSV:

```
Node.Title,external_id,custom_field,objective,value_reached
Wadus,133,campo-indicadores,100,"10,3"
```

- the external ID is valid
- but the rest of the CSV I think it's incomplete. I don't see the project has been updated, and there's no message informing the users

![Screenshot 2020-05-22 at 16 10 23](https://user-images.githubusercontent.com/17616/82676486-c41b7900-9c46-11ea-8272-0f3710e5d6ae.png)

entantoencuanto commented:
A validation message should appear there. The CSV is expected to contain `col_1 | col_2 | col_3 ... col_N` headers for the table contents. In this way multiple tables can be loaded with the same CSV. Other names are ignored. The form ignores the `objective` and `value_reached` columns and do nothing

ferblape commented:
Do you mean in your comment that you are going to review it?

entantoencuanto commented:
Yes, i'll add an alert if no valid columns are present

entantoencuanto committed:
Fix validations about table custom fields csv file format

entantoencuanto committed:
Add validation errors messages for plan_table_custom_fields_form to import_csv_form

entantoencuanto commented:
Fixed. Currently there was a validation error, but the error message were not being showed.

Also I've fixed the validation condition and added some translations.

I think now it's ready to merge

ferblape commented:
Normalize locales and merge

![Screenshot 2020-05-28 at 07 14 41](https://user-images.githubusercontent.com/17616/83101568-e849e080-a0b2-11ea-9d71-ddf5e06b6846.png)

entantoencuanto committed:
Normalize translations
